--- gtk+-2.11.1/gtk/gtkentry.c.set-invisible-char-to-bullet	2007-06-04 11:33:19.000000000 -0400
+++ gtk+-2.11.1/gtk/gtkentry.c	2007-06-04 13:52:40.000000000 -0400
@@ -537,7 +537,7 @@ gtk_entry_class_init (GtkEntryClass *cla
                                    g_param_spec_unichar ("invisible-char",
 							 P_("Invisible character"),
 							 P_("The character to use when masking entry contents (in \"password mode\")"),
-							 '*',
+							 (gunichar) 0x2022,
 							 GTK_PARAM_READWRITE));
 
   g_object_class_install_property (gobject_class,
@@ -1101,7 +1101,7 @@ gtk_entry_init (GtkEntry *entry)
 
   entry->editable = TRUE;
   entry->visible = TRUE;
-  entry->invisible_char = '*';
+  entry->invisible_char = (gunichar) 0x2022;
   entry->dnd_position = -1;
   entry->width_chars = -1;
   entry->is_cell_renderer = FALSE;
@@ -4495,7 +4495,7 @@ gtk_entry_get_visibility (GtkEntry *entr
  * gtk_entry_set_visibility() has been called to set text visibility
  * to %FALSE. i.e. this is the character used in "password mode" to
  * show the user how many characters have been typed. The default
- * invisible char is an asterisk ('*').  If you set the invisible char
+ * invisible char is a small bullet (Unicode character 2022).  If you set the invisible char
  * to 0, then the user will get no feedback at all; there will be
  * no text on the screen as they type.
  **/
