diff -rup atk-1.23.5/atk/Makefile.am atk-1.23.5-new/atk/Makefile.am
--- atk-1.23.5/atk/Makefile.am	2008-07-21 02:49:43.000000000 -0400
+++ atk-1.23.5-new/atk/Makefile.am	2008-09-09 11:25:26.000000000 -0400
@@ -139,7 +139,7 @@ s-enum-types-c: @REBUILD@ $(atk_headers)
 if OS_WIN32
 atk_def = atk.def
 libatk_1_0_la_LDFLAGS += -export-symbols $(srcdir)/atk.def -no-undefined -Wl,atk-win32-res.o
-libatk_1_0_la_DEPENDENCIES = atk-win32-res.o
+libatk_1_0_la_DEPENDENCIES = atk-win32-res.o atk.def
 
 install-def-file:
 	$(INSTALL) $(srcdir)/atk.def $(DESTDIR)$(libdir)/atk-1.0.def
diff -rup atk-1.23.5/atk/Makefile.in atk-1.23.5-new/atk/Makefile.in
--- atk-1.23.5/atk/Makefile.in	2008-07-21 02:50:09.000000000 -0400
+++ atk-1.23.5-new/atk/Makefile.in	2008-09-09 11:25:31.000000000 -0400
@@ -334,7 +334,7 @@ BUILT_SOURCES = atkmarshal.h atkmarshal.
 
 # ---------- Win32 stuff ----------
 @OS_WIN32_TRUE@atk_def = atk.def
-@OS_WIN32_TRUE@libatk_1_0_la_DEPENDENCIES = atk-win32-res.o
+@OS_WIN32_TRUE@libatk_1_0_la_DEPENDENCIES = atk-win32-res.o atk.def
 @MS_LIB_AVAILABLE_TRUE@noinst_DATA = atk-$(ATK_API_VERSION).lib
 EXTRA_DIST = atk.symbols atk.rc.in atkmarshal.list atkintl.h
 DISTCLEANFILES = \
