--- gtkmm-2.14.1/docs/reference/gtkmm-2.4.devhelp~	2008-08-02 12:16:32.000000000 +0200
+++ gtkmm-2.14.1/docs/reference/gtkmm-2.4.devhelp	2008-10-11 23:57:17.000000000 +0200
@@ -1,5 +1,5 @@
 <?xml version="1.0"?>
-<book xmlns:fo="http://www.w3.org/1999/XSL/Format" title="gtkmm 2.4 library Reference Manual" name="gtkmm 2.4" link="../../../doc/gtkmm-2.4/docs/reference/html/../../index.html">
+<book xmlns:fo="http://www.w3.org/1999/XSL/Format" title="gtkmm 2.4 library Reference Manual" name="gtkmm 2.4" link="../../../doc/gtkmm-2.4/docs/reference/html/index.html">
   <chapters>
     <sub name="Classes" link="../../../doc/gtkmm-2.4/docs/reference/html/classes.html">
       <sub name="Atk::Action" link="../../../doc/gtkmm-2.4/docs/reference/html/classAtk_1_1Action.html">
