--- csv.ml.orig	2007-05-29 13:07:33.000000000 +0100
+++ csv.ml	2007-05-29 13:07:44.000000000 +0100
@@ -45,7 +45,7 @@
 (* Uncomment the next line to enable Extlib's List function.  These
  * avoid stack overflows on really huge CSV files.
  *)
-(*open ExtList*)
+open ExtList
 
 type t = string list list
 
--- Makefile.orig	2007-05-29 13:11:36.000000000 +0100
+++ Makefile	2007-05-29 13:11:53.000000000 +0100
@@ -6,15 +6,15 @@
 # of file csv.ml.
 
 OCAMLCINCS	:=
-#OCAMLCINCS	:= -package extlib
+OCAMLCINCS	:= -package extlib
 OCAMLCFLAGS	:= -g
 OCAMLCLIBS	:=
-#OCAMLCLIBS	:= -linkpkg
+OCAMLCLIBS	:= -linkpkg
 
 OCAMLOPTINCS	:= $(OCAMLCINCS)
 OCAMLOPTFLAGS	:=
 OCAMLOPTLIBS	:=
-#OCAMLOPTLIBS	:= -linkpkg
+OCAMLOPTLIBS	:= -linkpkg
 
 OBJS		:= csv.cmo
 XOBJS		:= $(OBJS:.cmo=.cmx)
