diff -rup jpeg-6b/makefile.cfg jpeg-6b.new/makefile.cfg
--- jpeg-6b/makefile.cfg	2008-09-09 09:54:38.000000000 -0400
+++ jpeg-6b.new/makefile.cfg	2008-09-09 09:55:03.000000000 -0400
@@ -169,7 +169,7 @@ libjpeg.a: @A2K_DEPS@ $(LIBOBJECTS)
 # with libtool:
 libjpeg.la: @A2K_DEPS@ $(LIBOBJECTS)
 	$(LIBTOOL) --mode=link $(CC) $(CFLAGS) -o libjpeg.la $(LIBOBJECTS) \
-		-rpath $(libdir) -version-info $(JPEG_LIB_VERSION)
+		-no-undefined -rpath $(libdir) -version-info $(JPEG_LIB_VERSION)
 
 # sample programs:
 
