--- mingwrt-3.15.1-mingw32/mingwex/usleep.c	2008-05-04 13:15:41.000000000 +0100
+++ mingwrt-3.15.1-mingw32.usleep/mingwex/usleep.c	2008-10-16 17:58:36.000000000 +0100
@@ -26,6 +26,11 @@
 #define WIN32_LEAN_AND_MEAN
 #include <windows.h>
 
+/* RWMJ: This is defined in our local sys/types.h but for some reason
+ * isn't picked up.
+ */
+typedef unsigned int useconds_t;
+
 int __cdecl usleep(useconds_t useconds)
 {
     if(useconds == 0)
