diff -r 47135c91cd9c configure.ac
--- a/configure.ac	Thu Oct 09 11:23:09 2008 -0400
+++ b/configure.ac	Wed Oct 29 17:35:22 2008 +0000
@@ -116,11 +116,14 @@
 
 save_CFLAGS="$CFLAGS"
 save_LDFLAGS="$LDFLAGS"
+save_LIBS="$LIBS"
 CFLAGS="$CFLAGS $GTK_CFLAGS"
-LDFLAGS="$LDFLAGS $GTK_LIBS"
+LDFLAGS="$LDFLAGS $GTK_LDFLAGS"
+LIBS="$LIBS $GTK_LIBS"
 AC_CHECK_FUNC([gdk_cairo_create], [WITH_GTK_CAIRO=1], [WITH_GTK_CAIRO=0])
 CFLAGS="$save_CFLAGS"
 LDFLAGS="$save_LDFLAGS"
+LIBS="$save_LIBS"
 
 AC_DEFINE_UNQUOTED(WITH_GTK_CAIRO, [$WITH_GTK_CAIRO], [Whether to use cairo for GTK rendering])
 
