diff -r 47135c91cd9c src/gvnc.c
--- a/src/gvnc.c	Thu Oct 09 11:23:09 2008 -0400
+++ b/src/gvnc.c	Wed Oct 29 14:51:17 2008 +0000
@@ -380,7 +380,7 @@
 					ret = -1;
 				}
 			} else
-				ret = read(fd, gvnc->read_buffer, 4096);
+				ret = recv (fd, gvnc->read_buffer, 4096, 0);
 
 			if (ret == -1) {
 				switch (errno) {
@@ -439,9 +439,9 @@
 				ret = -1;
 			}
 		} else
-			ret = write(fd,
+			ret = send (fd,
 				    gvnc->write_buffer+offset,
-				    gvnc->write_offset-offset);
+				    gvnc->write_offset-offset, 0);
 		if (ret == -1) {
 			switch (errno) {
 			case EWOULDBLOCK:
