diff -r 47135c91cd9c src/vncdisplay.c
--- a/src/vncdisplay.c	Thu Oct 09 11:23:09 2008 -0400
+++ b/src/vncdisplay.c	Wed Oct 29 14:52:28 2008 +0000
@@ -1892,7 +1903,7 @@
 {
 	char file[PATH_MAX];
 	char sysdir[PATH_MAX];
-#ifndef __MINGW32__
+#ifndef WIN32
 	char userdir[PATH_MAX];
 	struct passwd *pw;
 	char *dirs[] = { sysdir, userdir };
@@ -1903,7 +1914,7 @@
 	strncpy(sysdir, SYSCONFDIR "/pki", PATH_MAX-1);
 	sysdir[PATH_MAX-1] = '\0';
 
-#ifndef __MINGW32__
+#ifndef WIN32
 	if (!(pw = getpwuid(getuid())))
 		return TRUE;
 
