--- openssl-0.9.7f/apps/CA.sh.ca-dir	2005-02-02 00:45:38.000000000 +0100
+++ openssl-0.9.7f/apps/CA.sh	2005-04-21 21:08:09.270233699 +0200
@@ -38,7 +38,7 @@
 VERIFY="$OPENSSL verify"
 X509="$OPENSSL x509"
 
-CATOP=./demoCA
+CATOP=../../CA
 CAKEY=./cakey.pem
 CACERT=./cacert.pem
 
--- openssl-0.9.7f/apps/CA.pl.ca-dir	2005-02-02 00:45:38.000000000 +0100
+++ openssl-0.9.7f/apps/CA.pl	2005-04-21 21:07:52.307995284 +0200
@@ -52,7 +52,7 @@
 $X509="$openssl x509";
 $PKCS12="$openssl pkcs12";
 
-$CATOP="./demoCA";
+$CATOP="../../CA";
 $CAKEY="cakey.pem";
 $CACERT="cacert.pem";
 
--- openssl-0.9.7f/apps/openssl.cnf.ca-dir	2005-03-30 12:20:17.000000000 +0200
+++ openssl-0.9.7f/apps/openssl.cnf	2005-04-21 21:08:29.581927172 +0200
@@ -34,7 +34,7 @@
 ####################################################################
 [ CA_default ]
 
-dir		= ./demoCA		# Where everything is kept
+dir		= ../../CA		# Where everything is kept
 certs		= $dir/certs		# Where the issued certs are kept
 crl_dir		= $dir/crl		# Where the issued crl are kept
 database	= $dir/index.txt	# database index file.
