--- gnutls-2.4.1.orig/src/Makefile.am	2008-06-19 11:00:10.000000000 +0100
+++ gnutls-2.4.1/src/Makefile.am	2008-09-05 09:11:59.000000000 +0100
@@ -65,7 +65,7 @@
 certtool_SOURCES = certtool.gaa certtool-gaa.h certtool-cfg.h	\
 	certtool-gaa.c certtool.c prime.c certtool-cfg.c
 certtool_LDADD = ../lib/libgnutls.la $(LIBGCRYPT_LIBS) \
-	$(LIBTASN1_LIBS) ../gl/libgnu.la @LTLIBREADLINE@
+	$(LIBTASN1_LIBS) ../gl/libgnu.la @LTLIBREADLINE@ -lintl
 if HAVE_LIBCFG
 certtool_LDADD += -lcfg+
 else
