diff -ur qt-win-opensource-src-4.4.3.orig/src/corelib/arch/qatomic_windows.h qt-win-opensource-src-4.4.3/src/corelib/arch/qatomic_windows.h
--- qt-win-opensource-src-4.4.3.orig/src/corelib/arch/qatomic_windows.h	2008-09-27 10:00:54.000000000 +0100
+++ qt-win-opensource-src-4.4.3/src/corelib/arch/qatomic_windows.h	2009-02-02 11:07:19.000000000 +0000
@@ -384,11 +384,11 @@
 #else
 
 extern "C" {
-    __declspec(dllimport) long __stdcall InterlockedCompareExchange(long *, long, long);
-    __declspec(dllimport) long __stdcall InterlockedIncrement(long *);
-    __declspec(dllimport) long __stdcall InterlockedDecrement(long *);
-    __declspec(dllimport) long __stdcall InterlockedExchange(long *, long);
-    __declspec(dllimport) long __stdcall InterlockedExchangeAdd(long *, long);
+    __declspec(dllimport) long __stdcall InterlockedCompareExchange(volatile long *, long, long);
+    __declspec(dllimport) long __stdcall InterlockedIncrement(volatile long *);
+    __declspec(dllimport) long __stdcall InterlockedDecrement(volatile long *);
+    __declspec(dllimport) long __stdcall InterlockedExchange(volatile long *, long);
+    __declspec(dllimport) long __stdcall InterlockedExchangeAdd(volatile long *, long);
 }
 
 inline bool QBasicAtomicInt::ref()
