diff -ur qt-win-opensource-src-4.4.3.orig/src/gui/inputmethod/qwininputcontext_win.cpp qt-win-opensource-src-4.4.3/src/gui/inputmethod/qwininputcontext_win.cpp
--- qt-win-opensource-src-4.4.3.orig/src/gui/inputmethod/qwininputcontext_win.cpp	2008-09-27 10:00:55.000000000 +0100
+++ qt-win-opensource-src-4.4.3/src/gui/inputmethod/qwininputcontext_win.cpp	2009-02-02 12:19:10.000000000 +0000
@@ -497,6 +497,9 @@
 }
 
 
+// http://lists.trolltech.com/qt-interest/2008-05/thread00666-0.html
+extern bool qt_sendSpontaneousEvent(QObject *, QEvent *);
+
 bool QWinInputContext::endComposition()
 {
     QWidget *fw = focusWidget();
