Index: src/libgdl/gdl-dock-master.c
===================================================================
--- src/libgdl/gdl-dock-master.c	(revision 20065)
+++ src/libgdl/gdl-dock-master.c	(working copy)
@@ -650,7 +650,11 @@
 
 #ifdef WIN32    
     GdkLineStyle lineStyle = GDK_LINE_ON_OFF_DASH;
+#if 0
     if (is_os_vista())
+#else /* RWMJ - missing */
+    if (0)
+#endif
     {
         // On Vista the dash-line is increadibly slow to draw, it takes several minutes to draw the tracking lines
         // With GDK_LINE_SOLID it is parts of a second
