Index: src/libnrtype/FontInstance.cpp
===================================================================
--- src/libnrtype/FontInstance.cpp	(revision 20075)
+++ src/libnrtype/FontInstance.cpp	(working copy)
@@ -443,7 +443,7 @@
         OUTLINETEXTMETRIC otm;
         GetOutlineTextMetrics(daddy->hScreenDC, sizeof(otm), &otm);
         GLYPHMETRICS metrics;
-        DWORD bufferSize=GetGlyphOutline (daddy->hScreenDC, glyph_id, GGO_GLYPH_INDEX | GGO_NATIVE | GGO_UNHINTED, &metrics, 0, NULL, &identity);
+        DWORD bufferSize=GetGlyphOutline (daddy->hScreenDC, glyph_id, GGO_GLYPH_INDEX | GGO_NATIVE, &metrics, 0, NULL, &identity);
         double scale=1.0/daddy->fontSize;
         n_g.h_advance=metrics.gmCellIncX*scale;
         n_g.v_advance=otm.otmTextMetrics.tmHeight*scale;
@@ -457,7 +457,7 @@
             doAdd=true;
         } else {
             std::auto_ptr<char> buffer(new char[bufferSize]);
-            if ( GetGlyphOutline (daddy->hScreenDC, glyph_id, GGO_GLYPH_INDEX | GGO_NATIVE | GGO_UNHINTED, &metrics, bufferSize, buffer.get(), &identity) <= 0 ) {
+            if ( GetGlyphOutline (daddy->hScreenDC, glyph_id, GGO_GLYPH_INDEX | GGO_NATIVE, &metrics, bufferSize, buffer.get(), &identity) <= 0 ) {
                 // shit happened
             } else {
                 // Platform SDK is rubbish, read KB87115 instead
