--- curl-7.18.2.orig/lib/setup.h	2008-04-23 21:07:52.000000000 +0100
+++ curl-7.18.2/lib/setup.h	2008-11-13 15:51:40.000000000 +0000
@@ -47,6 +47,12 @@
 
 #ifdef HAVE_CONFIG_H
 #include "config.h"
+
+#ifdef __MINGW32__ /* only for MinGW cross-compiler ... */
+/* We want the getaddrinfo functions. */
+#define _WIN32_WINNT 0x0501
+#endif /* __MINGW32__ cross-compiler */
+
 #else
 
 #ifdef _WIN32_WCE
