Remove functions that aren't supported on Win32 / PDCurses.

diff -ur mingw32-ocaml-curses-20020319.orig/curses/functions.c mingw32-ocaml-curses-20020319.mingw/curses/functions.c
--- mingw32-ocaml-curses-20020319.orig/curses/functions.c	2002-03-18 01:20:00.000000000 +0000
+++ mingw32-ocaml-curses-20020319.mingw/curses/functions.c	2008-11-15 23:01:46.000000000 +0000
@@ -449,7 +449,9 @@
 /* resize */
 
 ML3(wresize,err,window,int,int)
+#ifndef WIN32
 ML2(resizeterm,err,int,int)
+#endif
 
 /* scr_dump */
 
@@ -526,6 +528,8 @@
 ML2d(vidputs,err,chtype,(char->unit))
 BEG2 putc_function=ab;
   r_err(vidputs(a_chtype(aa),putc_callback)); END
+
+#ifndef WIN32
 ML2d(tparm,string,string,int array)
 BEG2 int t[10],i,n=Wosize_val(ab);
   if(n>10) n=10;
@@ -554,6 +558,7 @@
 ML1d(bool_terminfo_variable,string*string*string,int) BEG1 arrayret(bool) END
 ML1d(num_terminfo_variable,string*string*string,int) BEG1 arrayret(num) END
 ML1d(str_terminfo_variable,string*string*string,int) BEG1 arrayret(str) END
+#endif
 
 /* touch */
 
@@ -611,6 +616,7 @@
 END
 #undef ca
 
+#ifndef WIN32
 /* Du travail pour les esclaves de M$ */
 ML0d(winch_handler_on,unit)
 BEG0 signal(SIGWINCH,winch_handler); CAMLreturn(Val_unit); END
@@ -627,4 +633,4 @@
   ioctl(a_int(aa),TIOCGWINSZ,&ws);
   r_int_int(ws.ws_row,ws.ws_col);
 END
-
+#endif
diff -ur mingw32-ocaml-curses-20020319.orig/curses/ml_curses.c mingw32-ocaml-curses-20020319.mingw/curses/ml_curses.c
--- mingw32-ocaml-curses-20020319.orig/curses/ml_curses.c	2002-03-18 01:20:00.000000000 +0000
+++ mingw32-ocaml-curses-20020319.mingw/curses/ml_curses.c	2008-11-15 22:57:53.000000000 +0000
@@ -5,12 +5,14 @@
 #include <caml/fail.h>
 #include <stdio.h>
 #include <unistd.h>
-#include <ncurses.h>
+#include <curses.h>
 #include <term.h>
 /* Du travail pour les esclaves de M$ */
 #include <signal.h>
+#ifndef WIN32
 #include <termios.h>
 #include <sys/ioctl.h>
+#endif
 
 #define AWB(x) caml__dummy_##x=caml__dummy_##x; /* anti-warning bugware */
 
